### PR TITLE
Added recovery tests

### PR DESCRIPTION
**Pull request by susanxhuynh**
1. Kill one task (DN, NN, JN, scheduler).
2. Kill all tasks of one type.
3. Kill executors. Kill executors is commented out for now because it fails.

### CODE REVIEW

File: frameworks/hdfs/integration/tests/defaults.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import os
```
Comment (nickbp) [inline frameworks/hdfs/integration/tests/defaults.py]: Maybe just merge the contents of this file into task_utils.py? Doesn't feel like there's a lot of reason to separate them

File: frameworks/hdfs/TEST_README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Running test.sh
 Environmental Variables
 -----------------------
 CLUSTER_URL: set this variable to use a pre-existing DC/OS cluster. For example, CLUSTER_URL=https://33.175.150.150
+TEST_TYPES: which types of tests to run: sanity, recovery. For example, TEST_TYPES=recovery
```
Comment (nickbp) [inline frameworks/hdfs/TEST_README.md]: Maybe mention that the default is sanity

Also the filter should also support compound statements like `TEST_TYPES="sanity or recovery"`

**Comment by susanxhuynh** [comment]
@nickbp Thanks again for the review. I have made some changes based on your comments.

File: frameworks/hdfs/integration/tests/test_recovery.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    check_health()
+
+
+@pytest.mark.recovery
+def test_kill_all_namenodes():
+    for host in shakedown.get_service_ips(PACKAGE_NAME):
+        kill_task_with_pattern('NameNode', host)
+
+    check_health()
```
Comment (nickbp) [inline frameworks/hdfs/integration/tests/test_recovery.py]: missing ending newline

**Comment by nickbp** [comment]
LGTM aside from the above nitpicks

**Pull request merged**